cli/azd/pkg/templates/template.go

```diff
@@ -1,8 +1,30 @@
 package templates
 
 type Template struct {
-	Name           string `json:"name"`
-	DisplayName    string `json:"displayName,omitempty"`
-	Description    string `json:"description"`
+	// Name is the friendly short name of the template.
+	Name string `json:"name"`
+
+	// Description is a long description of the template.
+	Description string `json:"description"`
+
+	// Path is a fully qualified URI to a git repository,
+	// "{owner}/{repo}" for GitHub repositories,
+	// or "{repo}" for GitHub repositories under Azure-Samples (default organization).
+	Path string `json:"path"`
+}
+
+type ContractedTemplate struct {
+	Template
+
+	// RepositoryPath is always set to the same value as template.Path
+	// It is added for backwards compatibility with the old contract.
+	// This can be removed once the old contract is no longer used.
 	RepositoryPath string `json:"repositoryPath"`
```

Comment:
I don't understand this. Can you help me to digest it?  How's the backwards compatibility story here? 
My understanding is that the `template.json` is fused inside the azd binary, so older binaries are not affected after chafing the `template.json`.  That's why I don't get this

Reply:
@vhvb1989 VSCode, VS are going to use the output of `azd template --output json` to prompt for templates on their side.

Reply:
Created #2097 to track the final removal. I just want to formalize the sign-off so that we remove it at the right time, not at the time that's most convenient for us.

Reply:
I'll move it to contracts to make it clearer

Reply:
I don't feel strongly about the rename. I do feel the new name is better and less verbose, and I was slightly worried that `RepositoryPath` may not work be semantically correct in future worlds where it no longer points to a "repository" technically (perhaps stuff like #2038). This is more intuition than logical reasoning, so I'm happy if `RepositoryPath` resonates well with everyone.

Reply:
what @karolz-ms said :)

It is an implementation details for `azd` what is the name used for the `templates.json` where the source of truth is.  Customers are never expected to change this (only develoepers).
So, azd can manage mapping names internally and keep the output contracts as they currently are.

Anyway, thank you for explaining what this was about.  I understand it now.
I don't have a big concern about this.  So, whatever you decide I am happy with.

Reply:
I think I may have been overthinking this. Let's not rush to renames 😊. Keeping it as-is.